navapbc/template-infra
# 329 add email and external incident management tools integrations

## Ticket

https://github.com/navapbc/template-infra/issues/329

## Changes
incorporated email, VictorOps, and PagerDuty integrations into existing monitoring module. Additionally, secrets(integration URL) used for these integrations are securely stored using AWS Systems Manager (SSM).

## Context for reviewers
Integrations are actually subscriptions to existing SNS topic.
emails sns subscriptions take set of strings as input and creates subscription for every email.
https subscription fetches url of the endpoint for external incident management tool from the AWS SSM parameter. SSM parameter created with bash script to avoid storing secret in the code or in state.

## Testing
1. add email_alerts attribute to monitoring module definition and set value to set fo strings where every element is a email address. 
2. Run command `make infra-configure-monitoring-secrets APP_NAME={APP_NAME} ENVIRONMENT={ENV} URL={URL}`
3. apply terraform code `TF_CLI_ARGS_apply="-var=image_tag={TAG}" make infra-update-app-service APP_NAME={APP} ENVIRONMENT={ENV}`
4. Ensure email and https subscriptions were created with SNS topic with AWS UI. - passed

Email notification is received when alarm was triggered. 
<img width="1424" alt="image" src="https://github.com/navapbc/template-infra/assets/5009912/4580909e-961c-4e04-a05e-6049779d999d">
